src-java/kilda-persistence-tinkerpop/src/main/java/org/openkilda/persistence/ferma/repositories/FermaFlowRepository.java

```diff
@@ -593,6 +604,36 @@ public Collection<Flow> findLoopedByLoopSwitchId(SwitchId switchId) {
                 .collect(Collectors.toList());
     }
 
+    @Override
+    public Map<Integer, Collection<Flow>> findSwitchFlowsByPort(SwitchId switchId, Collection<Integer> ports) {
+        try {
+            @SuppressWarnings("unchecked")
+            Map<Integer, Collection<Flow>> portToFlowMap = collectToPortToFlowTreeMap(
+                    framedGraph().traverse(g -> g.V()
+                            .hasLabel(PathSegmentFrame.FRAME_LABEL)
+                            .or(
+                                    has(PathSegmentFrame.SRC_SWITCH_ID_PROPERTY, switchId.toString()),
+                                    has(PathSegmentFrame.DST_SWITCH_ID_PROPERTY, switchId.toString()))
+                            .as(PATH_SEGMENT_ALIAS)
+                            .in(FlowFrame.OWNS_PATHS_EDGE)
```

Comment:
Should it be `FlowPathFrame.OWNS_SEGMENTS_EDGE`?

Reply:
In fact, it is just a step in into owns edges, they are filtered later, on 620. The variable name is confusing a bit, maybe just make it `OWNS`, as it is actually: `String OWNS_PATHS_EDGE = "owns";`?

Reply:
yes, it works because we have 2 edges with name "owns". But, for example, what if we will decide to rename OWNS_SEGMENTS_EDGE from 'own' to something else? In both cases (leaving FlowFrame.OWNS_PATHS_EDGE or use "owns") we will lose connection between this code and FlowPathFrame.OWNS_SEGMENTS_EDGE

Reply:
yeah, I see, good point. I'll add another variable

Reply:
But the old value is still here